manager-rs/dauth-service/src/main.rs

```diff
@@ -67,6 +70,37 @@ fn build_context(dauth_opt: DauthOpt) -> Result<Arc<DauthContext>, DauthError> {
     }))
 }
 
+fn generate_keys(keyfile_path: &String) -> Keypair {
+    match fs::read(keyfile_path) {
+        Ok(keypair_bytes) => match Keypair::from_bytes(&keypair_bytes) {
+            Ok(keypair) => {
+                tracing::info!("Existing keyfile found");
+                keypair
+            }
+            Err(e) => {
+                tracing::warn!("Failed to parse existing key bytes, rebuilding -- {}", e);
+                build_keyfile(keyfile_path)
+            }
+        },
+        Err(e) => {
+            tracing::warn!("Failed to read content from '{}' -- {}", keyfile_path, e);
+            build_keyfile(keyfile_path)
+        }
+    }
+}
+
+fn build_keyfile(keyfile_path: &String) -> Keypair {
```

Comment:
Syntactically, I think the best to return here is a `Result<KeyPair, Err>` type than a `KeyPair` instance, that'd enable that the path prefix `create_dir_all()?` and `fs::write()` can return its own errors and gracefully fail in case of full disk or inability to create/write due to permission issues.